pydantic/type_adapter.py

```diff
@@ -2,16 +2,32 @@
 from __future__ import annotations as _annotations
 
 import sys
+from contextlib import contextmanager
 from dataclasses import is_dataclass
-from typing import TYPE_CHECKING, Any, Dict, Generic, Iterable, Set, TypeVar, Union, cast, final, overload
+from functools import cached_property, wraps
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Generic,
+    Iterable,
+    Iterator,
+    Set,
+    TypeVar,
+    Union,
+    cast,
+    final,
+    overload,
+)
 
 from pydantic_core import CoreSchema, SchemaSerializer, SchemaValidator, Some
 from typing_extensions import Literal, get_args, is_typeddict
 
 from pydantic.errors import PydanticUserError
 from pydantic.main import BaseModel
 
-from ._internal import _config, _generate_schema, _typing_extra
+from ._internal import _config, _generate_schema, _mock_val_ser, _typing_extra, _utils
```

Comment:
Same thing, let's just import the relevant classes from `_mock_val_ser`

Reply:
Elsewhere all the `MockValSer`s are coming via `_mock_val_ser.MockValSer`. So I think this is to avoid exporting out the internal class we should not expose. Went with same way here